packages/core/src/CSL/coreToCsl/coreToCsl.ts

```diff
@@ -283,3 +291,56 @@ export const tx = ({ body, witness, auxiliaryData }: Cardano.NewTxAlonzo): Trans
   // Possible optimization: only convert auxiliary data once
   return Transaction.new(txBody(body, auxiliaryData), txWitnessSet, txAuxiliaryData(auxiliaryData));
 };
+
+export const nativeScript = (script: Cardano.NativeScript): NativeScript => {
+  let cslScript: NativeScript;
+
+  switch (script.type) {
+    case Cardano.NativeScriptType.RequireSignature: {
+      cslScript = NativeScript.new_script_pubkey(
+        ScriptPubkey.new(Ed25519KeyHash.from_bytes(Buffer.from(script.keyHash, 'hex')))
+      );
+      break;
+    }
+    case Cardano.NativeScriptType.RequireAllOf: {
+      const cslScripts = NativeScripts.new();
+      for (const subscript of script.scripts) {
+        cslScripts.add(nativeScript(subscript));
+      }
+      cslScript = NativeScript.new_script_all(ScriptAll.new(cslScripts));
+      break;
+    }
+    case Cardano.NativeScriptType.RequireAnyOf: {
+      const cslScripts2 = NativeScripts.new();
+      for (const subscript of script.scripts) {
+        cslScripts2.add(nativeScript(subscript));
+      }
+      cslScript = NativeScript.new_script_any(ScriptAny.new(cslScripts2));
+      break;
+    }
+    case Cardano.NativeScriptType.RequireMOf: {
+      const cslScripts3 = NativeScripts.new();
+      for (const subscript of script.scripts) {
+        cslScripts3.add(nativeScript(subscript));
+      }
+      cslScript = NativeScript.new_script_n_of_k(ScriptNOfK.new(script.required, cslScripts3));
+      break;
+    }
+    case Cardano.NativeScriptType.RequireTimeBefore: {
+      cslScript = NativeScript.new_timelock_expiry(
+        TimelockExpiry.new_timelockexpiry(BigNum.from_str(script.slot.toString()))
+      );
+      break;
+    }
+    case Cardano.NativeScriptType.RequireTimeAfter: {
+      cslScript = NativeScript.new_timelock_start(
+        TimelockStart.new_timelockstart(BigNum.from_str(script.slot.toString()))
+      );
+      break;
+    }
+    default:
+      throw new InvalidNativeScript('Unknown script type');
```

Comment:
I think this error should be aligned with error in [cslToCore](https://github.com/input-output-hk/cardano-js-sdk/pull/377/files#diff-d44f980f3fe3aa875421d19871eb64f3c3481069f3fe6b499ee8891692364728R347)

Reply:
I think you are right, we can stick with SerializationError for consistency (and because it is literally a serialization error) and drop this custom error. We can always added it back when legitimately needed.

Reply:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/377/commits/b5a80d971f77d88933f07c51622f45ec4028b7b4